### ISSUE TITLE
`run()` fails when `dir` is a vector

### ISSUE THREAD

**Issue by kellijohnson-NOAA** [r4ss/r4ss#733]
https://github.com/r4ss/r4ss/blob/37b89fe4046fd271fa76548779e36b22276a0910/R/run.R#L80

If `dir` of `run()` is a vector the function fails because the previous line resets `dir` to be the directory of the loop. Thus, when it is on iteration two of the loop, the function fails because `dir` only has a length of one.

**Comment by iantaylor-NOAA** [comment]
Thanks @kellijohnson-NOAA, good catch. 
It looks like I used `dir[idir]` everywhere below that but forgot to delete that no-longer-needed line.

While fixing this, I tried to set up a test of the dir as vector option and went down a rabbit hole of cleaning up `populate_multiple_directories()` function to work with `check_exe()` and other recent changes. All that lead me to wonder the following.

**Question**: does it even make sense for `run()` to run on a vector of directories or for `populate_multiple_directories()` to exist, when we could just use {purrr} (or whatever) to map `run()` or `copy_SS_inputs()` to a vector of directories?
